### PR TITLE
Update OVN CI jobs to include new gathering of core dumps

### PR DESCRIPTION
**Pull request by astoycos**
Update CI jobs to include new gathering of core dumps
 
Follows PR https://github.com/openshift/must-gather/pull/173
    
Adds new step Registry chain `gather-core-dump` That will
gather core dumps off nodes
    
Activates gathering of Core Dumps for default E2E tests on
AWS,Azure,Gcp,and Vsphere and gathering of Network-logs
on Network specfic E2E tests
    
TODO: Activate in more Jobs?

### CODE REVIEW

**Comment by astoycos** [comment]
@trozet @dcbw

**Comment by dcbw** [comment]
@astoycos you want to put the coredump step before ipi-deprovision, otherwise the nodes will be gone and there won't be anyhting to get coredumps from

**Comment by dcbw** [comment]
@astoycos also:

```
ERROR: Run the following command to re-generate the registry metadata:
ERROR: $ make registry-metadata
```

**Comment by astoycos** [comment]
@dcbw ```[astoycos@localhost release]$ make registry-metadata
docker pull registry.svc.ci.openshift.org/ci/generate-registry-metadata:latest
Error response from daemon: Get "https://registry.svc.ci.openshift.org/v2/ci/generate-registry-metadata/manifests/latest": unauthorized: authentication required
make: *** [Makefile:59: registry-metadata] Error 1```

**Comment by dcbw** [comment]
> I just need to get access to openshift's quay.io right?

https://github.com/openshift/cluster-network-operator/wiki/Running-a-local-cluster-network-operator-for-plugin-development#get-the-openshift-ci-pull-secret

File: ci-operator/step-registry/gather/core-dump/OWNERS [hunk 1]
```diff
@@ -0,0 +1,11 @@
+approvers:
```
Comment (dcbw) [inline ci-operator/step-registry/gather/core-dump/OWNERS:1]: @astoycos you wrote this step, so add yourself to the OWNERS, and maybe tim and some others too. Clayton, wking, abhinav, at least probably don't need (or want) to be on here.

Comment (astoycos) [reply ci-operator/step-registry/gather/core-dump/OWNERS:1]: Sounds good

**Comment by trozet** [comment]
@dceara @astoycos @dcbw and we got a core dump for ovn-controller! However, I think we need the actual core as the stack trace doesn't reveal much I don't think:

           PID: 2777 (ovn-controller)
           UID: 0 (root)
           GID: 0 (root)
        Signal: 6 (ABRT)
     Timestamp: Fri 2020-08-28 02:23:56 UTC (13min ago)
  Command Line: ovn-controller unix:/var/run/openvswitch/db.sock -vfile:off --no-chdir --pidfile=/var/run/ovn/ovn-controller.pid -p /ovn-cert/tls.key -c /ovn-cert/tls.crt -C /ovn-ca/ca-bundle.crt -vconsole:info
    Executable: /usr/bin/ovn-controller
 Control Group: /kubepods.slice/kubepods-burstable.slice/kubepods-burstable-pod4c2a9090_42b0_4b4a_9f2d_9d562c27f018.slice/crio-8ae10644873f9db811350da6bd73570e2a525dfa30fe03c366c63691902c6fbd.scope
          Unit: crio-8ae10644873f9db811350da6bd73570e2a525dfa30fe03c366c63691902c6fbd.scope
         Slice: kubepods-burstable-pod4c2a9090_42b0_4b4a_9f2d_9d562c27f018.slice
       Boot ID: e997dddee4284ab3b5167eefa275fdda
    Machine ID: ec2d830afc87d088c9ab4e3c527307e8
      Hostname: ip-10-0-195-107
       Storage: /var/lib/systemd/coredump/core.ovn-controller.0.e997dddee4284ab3b5167eefa275fdda.2777.1598581436000000.lz4
       Message: Process 2777 (ovn-controller) of user 0 dumped core.
                
                Stack trace of thread 2777:
                #0  0x00007f7dcaffb387 n/a (/usr/lib64/libc-2.17.so)
                #1  0x00007ffca727d300 n/a (n/a)

**Comment by trozet** [comment]
We do have the DBs now @dceara if that helps at all to reproduce this.

**Comment by astoycos** [comment]
> @dceara @astoycos @dcbw and we got a core dump for ovn-controller! However, I think we need the actual core as the stack trace doesn't reveal much I don't think:
> 
> ```
>            PID: 2777 (ovn-controller)
>            UID: 0 (root)
>            GID: 0 (root)
>         Signal: 6 (ABRT)
>      Timestamp: Fri 2020-08-28 02:23:56 UTC (13min ago)
>   Command Line: ovn-controller unix:/var/run/openvswitch/db.sock -vfile:off --no-chdir --pidfile=/var/run/ovn/ovn-controller.pid -p /ovn-cert/tls.key -c /ovn-cert/tls.crt -C /ovn-ca/ca-bundle.crt -vconsole:info
>     Executable: /usr/bin/ovn-controller
>  Control Group: /kubepods.slice/kubepods-burstable.slice/kubepods-burstable-pod4c2a9090_42b0_4b4a_9f2d_9d562c27f018.slice/crio-8ae10644873f9db811350da6bd73570e2a525dfa30fe03c366c63691902c6fbd.scope
>           Unit: crio-8ae10644873f9db811350da6bd73570e2a525dfa30fe03c366c63691902c6fbd.scope
>          Slice: kubepods-burstable-pod4c2a9090_42b0_4b4a_9f2d_9d562c27f018.slice
>        Boot ID: e997dddee4284ab3b5167eefa275fdda
>     Machine ID: ec2d830afc87d088c9ab4e3c527307e8
>       Hostname: ip-10-0-195-107
>        Storage: /var/lib/systemd/coredump/core.ovn-controller.0.e997dddee4284ab3b5167eefa275fdda.2777.1598581436000000.lz4
>        Message: Process 2777 (ovn-controller) of user 0 dumped core.
>                 
>                 Stack trace of thread 2777:
>                 #0  0x00007f7dcaffb387 n/a (/usr/lib64/libc-2.17.so)
>                 #1  0x00007ffca727d300 n/a (n/a)
> ```

See https://github.com/openshift/must-gather/pull/173 for a fix to this

**Comment by knobunc** [comment]
/assign @stevekuznetsov

Steve -- Can you approve this please? We need to get the coredumps from the test runs so we can debug a problem when ovs is crashing in CI. Thanks

**Comment by abhat** [comment]
@petr-muller can you PTAL?

**Comment by astoycos** [comment]
Just FYI this test [failure](https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/openshift_release/11368/rehearse-11368-pull-ci-open-cluster-management-registration-operator-master-e2e/1306948752547778560) is fixed with https://github.com/openshift/must-gather/pull/179

**Review by stevekuznetsov — APPROVED** [review]

**Pull request merged**